int32 OPAQUE = 1
int32 TRANSPARENT = 2
int32 DUAL = 3
float32 dual_speed
int32 request_state_code
---
int32 SUCCESS = 1
int32 FAILED = 2
int32 response